doc/May-2022-Upgrade.md

```diff
@@ -0,0 +1,634 @@
+## Upgrading from an old version of kobo-docker (before May 2022)
+
+Current versions of kobo-docker require PostgreSQL 14, MongoDB 5 and Redis 6
+
+
+If you are running a version of kobo-docker that was last updated prior to
+May 2022 (i.e. commit TBC or older),
+you need to upgrade your databases prior to using the current version of
+kobo-docker (this repository) or
+[kobo-install](https://github.com/kobotoolbox/kobo-install).
+
+This is a step-by-step procedure to upgrade PostgreSQL and MongoDB.
+
+**This procedure has been tested on x86 architecture only.**
+
+### PostgreSQL
+
+**Be sure to have enough space left on the host filesystem before upgrading.**
+Check the size of the PostgreSQL database in  `.vols/db`, e.g. with
+`sudo du -hs .vols/db`, and ensure you have _more_ than this amount of space
+free.
+
+For this tutorial, we are using kobo-install to run docker-compose commands.
+If you do not use kobo-install, please replace `python run.py -cb` with `docker-compose -f docker-compose.primary.backend.template.yml -f docker-compose.primary.backend.yml [-f docker-compose.primary.backend.override.yml] [-f docker-compose.primary.backend.custom.yml]`
+
+1. Stop the containers
+
+    ```shell
+    user@computer:kobo-install$ python run.py --stop  
+    ```
+
+1. Edit composer file `docker-compose.primary.backend.template.yml`
+
+   - Temporarily, comment `postgis:14.2-3` to use PostgreSQL 9.5 with PostGIS 2.5  
+   - Add `- ./.vols/db14:/var/lib/postgresql/data14` below `- ./.vols/db:/var/lib/postgresql/data` 
+
+   It should look like this:
+
+   ```
+   # image: postgis/postgis:14-3.2
+   image: postgis/postgis:9.5-2.5
+    hostname: postgres
+    env_file:
+      - ../kobo-env/envfile.txt
+      - ../kobo-env/envfiles/databases.txt
+      - ../kobo-env/envfiles/aws.txt
+    volumes:
+      - ./.vols/db:/var/lib/postgresql/data
+      - ./.vols/db14:/var/lib/postgresql/data14
+   ```
+
+1. Run a one-off `postgres` container
+
+    ```shell
+    user@computer:kobo-install$ python run.py -cb run --rm postgres bash  
+    ```
+    
+1. Install PostgreSQL 14
+
+    ```shell
+    root@postgres:/# apt-get update
+    root@postgres:/# apt-cache policy postgresql-14-postgis-3
+    root@postgres:/# apt-cache policy postgis
+    ```
+
+    _Store the PostGIS version in a variable to use later_
+    
+    ```shell
+    root@postgres:/# POSTGIS_VERSION_14=$(apt-cache policy postgresql-14-postgis-3|grep Candidate:|awk '{print $2}')
+    root@postgres:/# apt-get install -y --no-install-recommends postgresql-14-postgis-3=${POSTGIS_VERSION_14} postgresql-14-postgis-3-scripts=${POSTGIS_VERSION_14} postgis postgresql-contrib-14
+    root@postgres:/# apt-get upgrade
+    ```
+
+1. Init DB
+
+    ```shell
+    root@postgres:/# chown -R postgres:postgres /var/lib/postgresql/data14/
+    root@postgres:/# su - postgres -c "/usr/lib/postgresql/14/bin/initdb -U $POSTGRES_USER --encoding=utf8 --locale=en_US.utf-8 -D /var/lib/postgresql/data14/"
+    ```
```

Comment:
Same comment as above re easy copying from the README. Maybe something like:

```
apt-get update && \
  apt-cache policy postgresql-14-postgis-3 && \
  apt-cache policy postgis
```

etc.

Reply:
(and the same comment for all the commands below 🙂)